Feature/sign in

davicruzof opened the pull request:
Tela de login com dois campo, sendo os mesmos para preenchimento do dados de acesso do usuário ( usuário e senha ) e uma botão para criação de uma nova conta, os inputs do campos são do tipo outlined.
Para implementação da mesma fiz o uso da lib styled-components para manipulação do style, ainda nessa feature foi adicionado as fontes da interface, tambem foi inserido a lib react-native-icons para utilização dos icones do app, ainda nessa feature foi configurado o lint e prettier.

davicruzof committed:
chore(asssets): Adicionando imagem de background sign in

davicruzof committed:
chore(Adicionado lib styled-components): Adicionando a lib usada para manipulação de estilos

davicruzof committed:
chore(asssets): Adicionando as tipografias do projeto DMSans e Roboto

davicruzof committed:
feat(textinput): Adicionando componente de textInput

davicruzof committed:
fix(textiput): Editando o componente para deixar dinamico

davicruzof committed:
chore(Adicionado lib react-native-vector-icons) lib de icons react native vector icons

davicruzof committed:
fix(textinput): Adicionando cor do placeholder e campo do tipo oculto

davicruzof committed:
feat(button): Adicionando componente de botão

davicruzof committed:
feat(signIn): Integração de componentes para montar a tela de login

davicruzof committed:
fix(signIn): Mudularizando componentes da tela de login

davicruzof committed:
chore(Adicionando eslint e prettier): adicionando regras de lint, utlizando como standard os padrões do Airbnb

davicruzof committed:
chore(Adicionado rules eslint): Adicionando regras de lint

davicruzof committed:
style(Font / Colors): Adicionando tipografia e cores

Adicionando cores e fontes em arquivos globais para melhor manutenção e utilização, como tambem sua escalabilidade

davicruzof committed:
fix(Font / Colors): Usando cores e tipografias goblais

Adicionando cores e fontes usando os arquivos utils

davicruzof committed:
fix(TextInput): Cor do placeholder do textinput

Alterando a cor do textInput para cor global

davicruzof committed:
style(Add color error inputs): Adicionando cor de error dos textinputs

davicruzof committed:
feat(Adicionando validador de campo): Adicionando texto de erro nos textinputs

davicruzof committed:
feat(Adicionando endpoint de login): Acesso login via endpoint utilizando fetch

Endpoint desenvolvido para utilizaçao em teste

davicruzof committed:
refactor(removendo console de debugger):

davicruzof committed:
fix(Coreção de cor): Corringindo cor do gradiente

FelipeNunes04 commented on src/components/Button/styles.js:
```diff
@@ -0,0 +1,20 @@
+import styled from 'styled-components';
+import { Colors } from '../../utils/Colors';
+import { Fonts } from '../../utils/Fonts';
```
Opa, configura path alias para seu projeto: https://christianlydemann.com/simpler-typescript-paths-with-path-aliases/
Esses imports com `../` não são uma boa prática

FelipeNunes04 commented on src/components/TextInput/index.js:
```diff
@@ -0,0 +1,59 @@
+  active,
+  errorInput = false, 
+  msgError = "",
+  ...rest }) => {
```
Você precisa tirar suas props, cria uma interface Props nesse arquivo mesmo.

FelipeNunes04 replied:
Todos as suas funções e métodos devem ter o tipo de retorno.
```suggestion
export const TextInput = ({
  isPassword = true,
  color,
  placeholder,
  label,
  active,
  errorInput = false, 
  msgError = "",
  ...rest }): ReactElement => {
```

FelipeNunes04 commented on src/components/TextInput/index.js:
```diff
@@ -0,0 +1,59 @@
+      }else{
+        setBorder(false)
+      }
+    }
```
Isso tá muito complexo, analisa como simplificar essa lógica.
```suggestion
    if(isFocused ) {
      setBorder(true)
    } else {
      setBorder(false)
    }
    
```

FelipeNunes04 commented on src/components/TextInput/index.js:
```diff
@@ -0,0 +1,59 @@
+        active={active}
+        {...rest}
+        placeholderTextColor={Colors.placeholder}
+        placeholder={isFocused ? "" : placeholder}
```
Esse tipo de lógica deve ficar em uma variável.
```suggestion
        const placeholder = isFocused ? "" : placeholder;
        ...
        placeholder={placeholder}
```

FelipeNunes04 commented on src/components/TextInputIcon/index.js:
```diff
@@ -0,0 +1,39 @@
+  msgError = "",
+  icon, 
+  isPassword, 
+  active, label, updateIcon, ...props }) => {
```
Aqui precisa cria uma interface para suas props também.

FelipeNunes04 replied:
Adicionar o tipo de retorno do seu componente.
```suggestion
export const TextInputIcon = ({ 
  errorInput = false, 
  msgError = "",
  icon, 
  isPassword, 
  active, label, updateIcon, ...props }): ReactElement => {
```

FelipeNunes04 commented on src/components/TextInput/styles.js:
```diff
@@ -0,0 +1,35 @@
+import styled from 'styled-components/native';
+import { Colors } from '../../utils/Colors';
+import { Fonts } from '../../utils/Fonts';
```
Substituir por Path Alias também.
```suggestion
import { Colors } from '@utils/Colors';
import { Fonts } from '@utils/Fonts';
```

FelipeNunes04 commented on src/components/TextInputIcon/index.js:
```diff
@@ -0,0 +1,39 @@
+            top: 24,
+            right: 20,
+            position: 'absolute',
+        }}
```
Usar styled components

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from 'react';
+import { Alert, StatusBar } from 'react-native';
+import bgLogin from '../../assets/bg-login.png';
```
```suggestion
import bgLogin from '@assets/bg-login.png';
```

FelipeNunes04 commented on src/components/TextInputIcon/index.js:
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import Icon from 'react-native-vector-icons/Feather';
+import { Colors } from '../../utils/Colors';
```
```suggestion
import { Colors } from '@utils/Colors';
```

FelipeNunes04 commented on src/components/TextInputIcon/index.js:
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import Icon from 'react-native-vector-icons/Feather';
+import { Colors } from '../../utils/Colors';
+import { TextInput } from '../TextInput';
```
```suggestion
import { TextInput } from '@components/TextInput';
```

FelipeNunes04 commented on src/components/TextInputIcon/index.js:
```diff
@@ -0,0 +1,39 @@
+        <TextInput
+            isPassword={isPassword}
+            label={label}
+            color="transparent"
```
Adicionar a cor transparente ao seu arquivo de cores e usar aqui

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+} from './style';
+import { Button } from '../../components/Button';
+import { TextInputIcon } from '../../components/TextInputIcon';
+import { Colors } from '../../utils/Colors';
```
```suggestion
import { Button } from '@components/Button';
import { TextInputIcon } from '@components/TextInputIcon';
import { Colors } from '@utils/Colors';
```

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+import { TextInputIcon } from '../../components/TextInputIcon';
+import { Colors } from '../../utils/Colors';
+
+export const SignIn = () => {
```
```suggestion
export const SignIn = (): ReactElement => {
```

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+        const myHeaders = new Headers();
+        myHeaders.append("Content-Type", "application/json");
+    
+        var requestOptions = {
```
Não é boa prática utilizar `var` no código
```suggestion
        const requestOptions = {
```

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+        .then(response => response.json())
+        .then(result => {
+            result.message === 'Usuario inválido' ? setErrorUser(true) : setErrorUser(false);
+            result.message === 'Senha inválida' ? setErrorPassword(true) : setErrorPassword(false);
```
Não é recomendado usar strings hard coded no código, o ideal é coloca-las em constantes ou em arquivos de tradução.

FelipeNunes04 replied:
Não se pode misturar a camada de View com a camada que faz comunicação com apis estercas, você tem que separar isso na sua arquitetura.

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+    },[password])
+
+
+    const login = async() => {
```
```suggestion
    const login = async (): ReactElement => {
```

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+                            errorInput={errorUser}
+                            msgError="Usuário inválido"
+                            isPassword={!passwordView}
+                            icon={userView ? 'eye-off' : 'eye'}
```
Esse tipo de lógica deve estar em uma variável.
```suggestion
                            const textInputIcon = userView ? 'eye-off' : 'eye';
                            ...
                            icon={textInputIcon}
```

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+                            msgError="Senha incorreta"
+                            errorInput={errorPassword}
+                            isPassword={!passwordView}
+                            icon={passwordView ? 'eye-off' : 'eye'}
```
Esse tipo de lógica deve estar em uma variável.

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+                            errorInput={errorPassword}
+                            isPassword={!passwordView}
+                            icon={passwordView ? 'eye-off' : 'eye'}
+                            active={password.length > 0}
```
Esse tipo de lógica deve estar em uma variável.

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+                            onPress={login}
+                            text="Entrar"
+                            active={activeButton}
+                            style={{ marginTop: 49}}
```
Todos os styles devem ficar no arquivos de styles

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+                                        setActiveRegisterButton(
+                                            !activeRegisterButton,
+                                        )
+                                    }
```
Extrair essa lógica para uma função.

FelipeNunes04 commented on src/views/SignIn/style.js:
```diff
@@ -0,0 +1,77 @@
+import styled from "styled-components";
+import LinearGradient from 'react-native-linear-gradient'
+import { Fonts } from "../../utils/Fonts";
+import { Colors } from "../../utils/Colors";
```
```suggestion
import { Fonts } from "@utils/Fonts";
import { Colors } from "@utils/Colors";
```

FelipeNunes04 commented on src/views/SignIn/index.js:
```diff
@@ -0,0 +1,132 @@
+        });
+
+        const myHeaders = new Headers();
+        myHeaders.append("Content-Type", "application/json");
```
Aqui tem muita lógica, você pode criar outros métodos para simplificar a leitura do código

FelipeNunes04 commented on src/components/Label/styles.js:
```diff
@@ -0,0 +1,32 @@
+import styled from 'styled-components/native';
+import { Colors } from '../../utils/Colors';
+import { Fonts } from '../../utils/Fonts';
```
```suggestion
import { Colors } from '@utils/Colors';
import { Fonts } from '@utils/Fonts';
```

davicruzof committed:
fix:(sign in): corrigindo state de visualização de usuário digitado

FelipeNunes04 commented on src/components/Button/index.js:
```diff
@@ -0,0 +1,11 @@
+
+import { Container, TextButton } from './styles';
+
+export const Button = ({ text, ...props }) => {
```
Suas props devem ser tinadas e seu componente tem que indicar qual o retorno dele. Pode criar uma interface Props nesse arquivo mesmo.
```suggestion
export const Button = ({ text, ...props }: Props): ReactElement => {
```

davicruzof commented:
![](https://github.trello.services/images/mini-trello-icon.png) [View Sign In](https://trello.com/c/ak9aZwnd/4-view-sign-in)

davicruzof committed:
refactor(typescript): Adicionand typescript

Adicionado ao projeto path alias como tambem com linguagem typescript

davicruzof committed:
refactor(reviwer solicitação): Refatoração das solicitações do review

Adicionado texto estaticos em constantes, adicionando os styles ao padrão utilizando styled-components, separando requisição a api do arquivo de interface

FelipeNunes04 commented on App.tsx:
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { SignIn } from './src/views/SignIn';
```
Opa @davicruzof, não é legal ter a pasta `src` nos seus imports, ajusta seu alias para importar a partir de `views`

FelipeNunes04 commented on src/components/Button/index.tsx:
```diff
@@ -0,0 +1,17 @@
+import { Container, TextButton } from './styles';
+
+interface TypeButtonProps extends ButtonProps {
+    text: string | '';
+    active: Boolean | false;
```
Aqui você não precisa definir esses valores padrões

FelipeNunes04 commented on src/components/IconButton/styles.tsx:
```diff
@@ -0,0 +1,16 @@
+import styled from 'styled-components/native';
+import { Colors } from '@utils/Colors';
+import Feather from 'react-native-vector-icons/Feather';
```
Não é uma boa prática importar arquivos do projeto antes de imports de bibliotecas externas

FelipeNunes04 commented on src/components/Input/index.tsx:
```diff
@@ -0,0 +1,60 @@
+import { Container, ContainerInput } from './styles';
+
+interface InputProps extends TextInputProps {
+    status: boolean | false;
+    text: string | '';
+    error: string | '';
+    errorInput: boolean | false;
```
Aqui também não precisa desses valores

FelipeNunes04 commented on src/components/Input/index.tsx:
```diff
@@ -0,0 +1,60 @@
+    }, [isFocused]);
+
+    useEffect(() => {
+        setShowLabel(!isFocused && !status ? false : true);
```
Lógicas assim não podem ficar dentro do render, o recomendado é colocar em uma variável ou método

FelipeNunes04 commented on src/services/login.tsx:
```diff
@@ -0,0 +1,28 @@
+        })
+        .catch((error) => reject(error));
+    })
+};
```
Adiciona uma linha em branco aqui

FelipeNunes04 commented on src/views/SignIn/index.tsx:
```diff
@@ -0,0 +1,135 @@
+import { Button } from '@components/Button';
+import { Alert, StatusBar } from 'react-native';
+import { Colors } from '@utils/Colors';
+import { apiRequest } from '../../services/login';
```
Ajusta esse import também para usar um alias para Services

FelipeNunes04 commented on src/views/SignIn/index.tsx:
```diff
@@ -0,0 +1,135 @@
+
+        const res = await apiRequest(raw);
+
+        setErrorUser(res.message === msgErrorUser ? true : false);
+        setErrorPassword(res.message === msgErrorPassword ? true: false);
```
Não é legal colocar essa lógica na chamada do método, coloca em uma linha antes, deixa o código mais legível

FelipeNunes04 commented on src/views/SignIn/styles.tsx:
```diff
@@ -0,0 +1,85 @@
+
+export const ContainerButton = styled.View`
+    margin-top: 40px;
+`;
```
Faltou uma linha em branco no final do arquivo

FelipeNunes04 commented on src/views/SignIn/index.tsx:
```diff
@@ -0,0 +1,135 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { IconButton } from '@components/IconButton';
+import { Input } from '@components/Input';
+import {
+    Title,
+    Subtitle,
+    Background,
+    FromContainer,
+    RegisterTitle,
+    RegisterButton,
+    ImageBackground,
+    RegisterContainer,
+    RegisterButtonText,
+    ContainerButton
+} from './styles';
+import bg from '@assets/bgLogin.png';
+import { Button } from '@components/Button';
+import { Alert, StatusBar } from 'react-native';
+import { Colors } from '@utils/Colors';
```
Você precisa organizar seus imports, coloca primeiro bibliotecas externas e depois importa seu código do projeto. Separa com uma linha em branco

FelipeNunes04 commented on src/components/Label/styles.ts:
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components/native';
+import { Colors } from '@utils/Colors';
```
Para organizar melhor, você pode colocar uma linha em branco entre esses imports

FelipeNunes04 commented on src/components/Input/styles.tsx:
```diff
@@ -0,0 +1,28 @@
+import { Colors } from '@utils/Colors';
+import { Fonts } from '@utils/Fonts';
+import styled from 'styled-components/native';
```
Esse import deve ser o primeiro do arquivo.

FelipeNunes04 replied:
opa @davicruzof, faltou ajustar o import aqui

davicruzof committed:
refactor(path): Ajustando path para ter como base a pasta scr

Solicitaçõ: "Opa @davicruzof, não é legal ter a pasta src nos seus imports, ajusta seu alias para importar a partir de views"

davicruzof committed:
refactor(tipagem): Removendo valores padrões

davicruzof committed:
refactor(imports): Organizando imports

davicruzof committed:
refactor(extensão): Ajustando extensão de js para tsx

davicruzof committed:
refactor(libs): Ajustando libs

davicruzof committed:
refactor(imports): Ajustando imports

davicruzof committed:
Rename login.tsx to Login.tsx

FelipeNunes04 commented on src/views/SignIn/index.tsx:
```diff
@@ -0,0 +1,138 @@
+    ContainerButton
+} from './styles';
+
+import bg from '@assets/bgLogin.png';
```
Dá uma olhada nesse link para conseguir importar imagens: https://stackoverflow.com/questions/52759220/importing-images-in-typescript-react-cannot-find-module

FelipeNunes04 commented on src/views/SignIn/index.tsx:
```diff
@@ -0,0 +1,138 @@
+                        >
+                            <Input
+                                error="Senha incorreta"
+                                errorInput={errorPassword}
+                                status={!!password}
+                                text="Senha"
+                                secureTextEntry={!passwordView}
+                                value={password}
+                                onChangeText={(value: string) =>
+                                    setPassword(value)
+                                }
+                            />
+                        </IconButton>
+                        <ContainerButton>
+                            <Button
+                                onPress={login}
+                                text="Entrar"
+                                active={activeButton}
+                                title={'Entrar'}
+                            />
+                        </ContainerButton>
+                        <RegisterContainer>
+                            <RegisterTitle>
+                                Ainda não é cadastrado?
+                            </RegisterTitle>
+                            <RegisterButton>
+                                <RegisterButtonText
+                                    active={activeRegisterButton}
+                                    onPress={register}
+                                >
+                                    Inscreva-se
+                                </RegisterButtonText>
+                            </RegisterButton>
+                        </RegisterContainer>
+                    </FromContainer>
+                </Background>
+            </Background>
+        </ImageBackground>
+    );
+};
```
Aqui você tem sua lógica de login misturada com as telas, o ideal seria ter uma camada de lógica e uma camada onde você desenha seu layout.

davicruzof committed:
refactor(arquitetura): Alteração na arquitetura do projeto

Ajustando a arquitetura com módulos independentes

davicruzof committed:
refactor(imports): Ajustando imports
